### PR TITLE
QA-468: add launch submission on mc that doesn't have values for inputs

### PR DESCRIPTION
**Pull request by ansingh7115**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import org.scalatest.{FreeSpecLike, Matchers}
+import org.broadinstitute.dsde.workbench.fixture._
+import spray.json._
+import DefaultJsonProtocol._
+import org.scalatest.time.{Minutes, Seconds, Span}
+import org.scalatest.concurrent.Eventually
+
+
+class MethodLaunchSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with Matchers
+  with CleanUp with RandomUtil with Retry with Eventually
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala]: the `CleanUp` fixture is not used in test.

File: automation/test-reports/TEST-org.scalatest.DeferredAbortedSuite.xml [hunk 2]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (ansingh7115) [inline automation/test-reports/TEST-org.scalatest.DeferredAbortedSuite.xml]: Woah, this shouldn't have been committed

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        withMethod("MethodLaunchSpec_abort", MethodData.SimpleMethod) { methodName =>
+          val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+          Rawls.methodConfigs.createMethodConfigInWorkspace(billingProject, workspaceName,
+            method, method.methodNamespace, method.methodName, 1,
+            SimpleMethodConfig.inputs, SimpleMethodConfig.outputs, method.rootEntityType)
+
+          val submissionId = Rawls.submissions.launchWorkflow(billingProject, workspaceName, method.methodNamespace, method.methodName, method.rootEntityType, "participant1", "this", false)
+
+          Rawls.submissions.abortSubmission(billingProject, workspaceName, submissionId)
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala:68]: I would check submission status first before abort. Sometimes, new submission status is `Error` instead `Submitted`, test should just fail here, not continued.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+class MethodLaunchSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with Matchers
+  with CleanUp with RandomUtil with Retry with Eventually
+  with BillingFixtures with WorkspaceFixtures with MethodFixtures {
+
+  val methodConfigName: String = SimpleMethodConfig.configName + "_" + UUID.randomUUID().toString
+  val operations = Array(Map("op" -> "AddUpdateAttribute", "attributeName" -> "participant1", "addUpdateAttribute" -> "testparticipant"))
+  val entity: Array[Map[String, Any]] = Array(Map("name" -> "participant1", "entityType" -> "participant", "operations" -> operations))
+
+  "launch workflow with input not defined" in {
```
Comment (marctalbott) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala]: It'd be nice for the test name to make it clear that launching a workflow with undefined input will throw an exception

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala (hunk 3 above)
Comment (ansingh7115) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala:68]: good point!

**Review by marctalbott — APPROVED** [review]

**Comment by ansingh7115** [comment]
jenkins retest

**Review by aweng98 — APPROVED** [review]
👍

**Comment by ansingh7115** [comment]
jenkins retest

**Comment by ansingh7115** [comment]
jenkins retest

**Pull request merged**